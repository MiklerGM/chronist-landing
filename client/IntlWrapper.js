import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
