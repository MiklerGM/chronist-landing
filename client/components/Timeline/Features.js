import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import featureListRU from './TimelineDataRU';
import featureListEN from './TimelineDataEN';

import './Features.less';

class Feature extends React.Component {
  render() {
    return (
      <div
        className='feature__img'
        onClick={() => this.props.change(this.props.data.id)}
        onMouseOver={() => this.props.change(this.props.data.id)}
      >
        <img
          src={this.props.data.img}
          alt={this.props.data.name}
          className={`img-responsive img-feature ${this.props.data.inactive ? 'inactive' : ''}`}
        />
      </div>
    );
  }
}

const Preview = ({ id, featureList }) => (
  <div className='feature__preview'>
    <ReactCSSTransitionGroup
      className="feature__map"
      component="div"
      transitionName={{
        enter: 'enter',
        leave: 'leave'
      }}
      transitionEnterTimeout={500}
      transitionLeaveTimeout={3000}
      transitionAppear={false}
    >
      <img
        className='feature__over'
        key={featureList[id].id}
        src={featureList[id].pic}
        alt={featureList[id].name}
      />
    </ReactCSSTransitionGroup>
    <div className='feature--description'>
      <p><strong>{featureList[id].name}</strong></p>
      <p>{featureList[id].desc}</p>
    </div>
  </div>
);

class Features extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: 0,
      // direction: 'left'
    };
  }

  changeId(id) {
    this.setState({ id });
  }

  swap(id) {
    this.setState({ id });
  }

  next() {
    this.setState(state => ({
      id: (state.id + 1 < featureListEN.length) ? state.id + 1 : 0
    }));
  }

  prev() {
    this.setState(state => ({
      id: (this.state.id - 1 < 0) ? featureListEN.length - 1 : this.state.id - 1
      // direction:  'left'
    }));
  }

  render() {
    const featureList = (this.props.locale === 'ru') ? featureListRU : featureListEN;
    return (
      <div className='page--segment'>
        <div className='page--content'>
          <h2>
            <span><FormattedMessage id='feature.title' /></span>
          </h2>
          <div className='feature__container'>
            <Preview
              id={this.state.id}
              featureList={featureList}
            />
            <div className='feature__wrapper'>
              {featureList.map((feature, id) => (
                <Feature
                  data={feature}
                  key={feature.name}
                  id={id}
                  change={v => this.changeId(v)}
                />))
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Features.propTypes = {
  locale: PropTypes.string.isRequired
};

Preview.propTypes = {
  id: PropTypes.number.isRequired,
  featureList: PropTypes.array.isRequired
};

Feature.propTypes = {
  change: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired
};

export default Features;
