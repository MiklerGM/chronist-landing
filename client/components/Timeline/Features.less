.img-responsive {
  width: 33%;
  display: inline-block;
  float: left;
  max-width: 120px;
  max-height: 120px;
}

.feature__container {
  // max-width: 1040px;
  // margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  // grid-template-rows
}

.feature__wrapper {
  grid-column: span 6;
  @media screen and (max-width: 990px) {
    grid-column: span 12;
  }
}

.feature__map {
  position: relative;
  height: 400px;
  // height: auto;
  // overflow: hidden;
  max-width: 700px;
  max-height: 400px;
  left:0;
  right:0;
  margin-left:auto;
  margin-right:auto;
  & > img  {
    position: absolute;
  }
  // & > div {
  //   position: absolute;
  // }
}

.feature__preview {
  position: relative;
  grid-column: span 6;
  // padding-top: ;
  min-height: 480px;
  @media screen and (max-width: 990px) {
    padding-top: 20px;
    text-align: center;
    grid-column: span 12;
    height: 450px;
  }

  & img {
    // position: relative;
    top: 25px;
    // left: 25px;
    left:0;
    right:0;
    margin-left:auto;
    margin-right:auto;
    padding-bottom: 40px;
    width: 450px;
    max-height: 400px;
    @media screen and (max-width: 990px) {width: 98%;}
  }

  & .feature__over {
    z-index: 3;
    opacity: 1;
  }

  & .feature__under {
    z-index: 1;
    opacity: 1;
  }

  & .feature--description {
    // clear: both;
    // position: relative;

    // top: 330px;
    // left:0;
    // right:0;
    // margin-left:auto;
    // margin-right:auto;
  }

  .enter {
    opacity: 0.01;
  }

  .enter.enter-active {
    opacity: 1;
    transition: opacity 500ms ease;
  }

  .leave {
    opacity: 1;
  }

  .leave.leave-active {
    opacity: 0.01;
    transition: opacity 3000ms ease-out;
  }
}

.overlay {
  cursor: pointer;
  position: fixed;
  overflow: hidden;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 5;
  background-color: rgba(45, 55, 64, 0.9);
  text-align: center;
  max-height: 100%;
}

.video--container {
  display: block;
  width: 100vw;
  height: 65vh;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  overflow: hidden;
  text-align: center;
  margin: 0 auto;
}

.overlay--control {
  width: 50%;
  height: 65vh;
  transform: translate(0, -50%);
  top: 50vh;
  color: white;
  position: fixed;
  z-index: 10;

  i {
    position: relative;
    color: gray;
    display: inline-block;
    font-style: normal;
    font-size: 25px;
    padding: 15px;
    line-height: inherit;
    top: 25%;
    user-select: none;
    background-color: rgba(0, 0, 0, 0.3);
    vertical-align: middle;
  }
}

.overlay--control__right {
  right: 0;
  margin-right: 0;
  left: auto;
  & i { float: right; }

  &:hover {
    & i { color: white; }
  }
}

.overlay--control__left {
  // background: linear-gradient(to left, rgba(0,0,0,0),#000000 60%);
  left: 0;
  margin-left: 0;
  right: auto;

  & i { float: left; }

  &:hover i { color: white; }
}

.overlay--control__close {
  top: 0;
  right: 0;
  float: right;
  display: inline-block;
  color: gray;
  font-style: normal;
  user-select: none;
  font-size: 25px;
  padding: 15px;

  & i { font-style: normal; }

  &:hover { color: white; }
}

.item {
  position: absolute;
  margin: auto;
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: auto;
  background-repeat: no-repeat;
  background-size: contain;

  & img {
    height: 400px;
    background-repeat: no-repeat;
    background-size: contain;
  }
}

.item--description {
  background-color: rgba(0, 0, 0, 0.1);
  color: #fff;
  padding: 5px;

  & p { color: #fff; }
}

.video--container {
  .enter-left {
    transform: translate(100vw);
  }
  .enter-left.enter-left-active {
    transform: translate(0);
    transition: transform 300ms ease-in-out;
  }
  .leave-left {
    transform: translate(0);
  }
  .leave-left.leave-left-active {
    transform: translate(-100vw, 0);
    transition: transform 300ms ease-in-out;
  }

  .enter-right {
    transform: translate(-100vw);
  }
  .enter-right.enter-right-active {
    transform: translate(0%);
    transition: transform 300ms ease-in-out;
  }
  .leave-right {
    transform: translate(0%);
  }
  .leave-right.leave-right-active {
    transform: translate(100vw);
    transition: transform 300ms ease-in-out;
  }
}

.fa-play-circle-o {
  opacity: 0;
  animation: fade 1s linear;
}

@keyframes fade {
  0%,100% { opacity: 1 }
  50% { opacity: 0 }
}


