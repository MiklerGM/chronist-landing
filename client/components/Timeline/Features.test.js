import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import Features from './Features';

const TestComponent = () => (
  <IntlWrapper>
    <Features
      locale='ru'
    />
  </IntlWrapper>
);

// it must render
it('Features render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Features render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Features Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
