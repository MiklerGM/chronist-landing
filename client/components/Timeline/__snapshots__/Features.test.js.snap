// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Snapshot 1`] = `
<div
  className="page--segment"
>
  <div
    className="page--content"
  >
    <h2>
      <span>
        Функционал
      </span>
    </h2>
    <div
      className="feature__container"
    >
      <div
        className="feature__preview"
      >
        <div
          className="feature__map"
        >
          <img
            alt="Политические границы"
            className="feature__over"
          />
        </div>
        <div
          className="feature--description"
        >
          <p>
            <strong>
              Политические границы
            </strong>
          </p>
          <p>
            Изменения политических границ от античности до наших дней
          </p>
        </div>
      </div>
      <div
        className="feature__wrapper"
      >
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Политические границы"
            className="img-responsive img-feature "
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Изобретения"
            className="img-responsive img-feature "
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Исследование данных"
            className="img-responsive img-feature "
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Экспорт данных"
            className="img-responsive img-feature "
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Личные сценарии"
            className="img-responsive img-feature "
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Войны"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Население"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Религии"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Языки"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Нации"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Катастрофы"
            className="img-responsive img-feature inactive"
          />
        </div>
        <div
          className="feature__img"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Экспедиции"
            className="img-responsive img-feature inactive"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
